### PR TITLE
Add GitHub actions workflow

### PR DESCRIPTION
**Pull request by christopher-mohr**
This adds a first simple GitHub Actions workflow for running tests and building the project on every push/PR. 

The `testcontainers` version had to be updated since this version includes a fix for a docker image retrieval issue (https://github.com/testcontainers/testcontainers-java/issues/3574) that occurred in the GitHub action workflow run.

### CODE REVIEW

File: .github/workflows/ci_test_build_package.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# # This workflow runs all available tests and builds the .jar files
+name: Varianstore CI
```
Comment (sven1103) [inline .github/workflows/ci_test_build_package.yml]: Note: A descriptive name what the workflow actually does. This name will pop up in the workflow overview once it runs.

File: .github/workflows/ci_test_build_package.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      - uses: actions/checkout@v2
+      # set up JDK 1.8
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+      # build with maven and run tests
+      - name: Build with Maven and run tests
+        run: mvn --batch-mode verify
```
Comment (sven1103) [inline .github/workflows/ci_test_build_package.yml]:
```suggestion
        run: mvn --batch-mode verify
        
```

**Comment by sven1103** [comment]
If you add caching, then you can speed up your build significantly:

```
- name: Load local Maven repository cache
        uses: actions/cache@v2
        with:
          path: ~/.m2/repository
          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
          restore-keys: |
            ${{ runner.os }}-maven-
```

**Comment by christopher-mohr** [comment]
Thanks for the review @sven1103! Changed the two things accordingly and added the Maven repository caching.

**Review by sven1103 — APPROVED** [review]

**Pull request merged**